packages/client/src/app/pages/portfolio/blogs/blogs.component.html

```diff
@@ -0,0 +1,113 @@
+<div class="tools" *ngIf="currentUser && currentUserIsSame()">
+    <div class="tool-buttons">
+        <ng-container *ngIf="listView; else galleryView">
+            <button matRipple class="tool left" (click)="switchView()"><i-feather name="grid"></i-feather>Gallery</button>
+            <button matRipple class="tool mid" (click)="openNewBlogForm()"><i-feather name="plus"></i-feather>Blog</button>
+            <button matRipple class="tool end"><i-feather name="filter"></i-feather>Filter</button>
+        </ng-container>
+        <ng-template #galleryView>
+            <button matRipple class="tool" (click)="switchView()"><i-feather name="list"></i-feather>List</button>
+        </ng-template>
+    </div>
+    <form [formGroup]="searchBlogs" class="tool-search" (ngSubmit)="searchFor()" *ngIf="listView">
+        <div class="search-bar">
+            <input type="text" formControlName="query" class="search-input" placeholder="Search blogs...">
+            <button matRipple class="search-submit" type="submit"><i-feather name="search"></i-feather></button>
+        </div>
+    </form>
+</div>
+
+<ng-container *ngIf="listView; else showGallery">
+    <ng-container *ngIf="userBlogsData !== null">
+        <div class="blog-list" *ngIf="userBlogsData.totalDocs !== 0; else noBlogs">
```

Comment:
Aren't these two if-checks kind of the same thing? Can a user have non-null userBlogsData, but still have zero totalDocs?

Reply:
in the resolver for this page, if there is no currently logged in user, the observable it returns has a null value for userBlogsData

this check is to make sure it isn’t null, so it doesn’t throw up errors when non-users and users who aren’t the owner of the portfolio navigate to this page